aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+    // Use the alternative limits for governance proposals
+    // TODO: In the future, consider adding the execution hashes of the scripts to the approval list.
+    txn_gas_params.max_execution_gas = txn_gas_params.max_execution_gas_gov;
+    txn_gas_params.max_io_gas = txn_gas_params.max_io_gas_gov;
+    txn_gas_params.max_storage_fee = txn_gas_params.max_storage_fee_gov;
+    state_view.set_state_value(
+        StateKey::resource(&AccountAddress::ONE, &StructTag {
+            address: AccountAddress::ONE,
+            module: Identifier::new(GasScheduleV2::MODULE_IDENTIFIER).unwrap(),
+            name: Identifier::new(GasScheduleV2::TYPE_IDENTIFIER).unwrap(),
+            type_args: vec![],
+        })?,
+        StateValue::new_legacy(
+            bcs::to_bytes(&GasScheduleV2 {
+                feature_version: gas_feature_version,
+                entries: gas_params.to_on_chain_gas_schedule(gas_feature_version),
+            })?
+            .into(),
+        ),
+    );
+
+    let chain_id = ChainIdResource::fetch_config(&state_view)
+        .ok_or_else(|| anyhow!("failed to fetch chain id"))?;
+
+    // Execute the governance scripts
+    let resolver = state_view.as_move_resolver();
+
+    const DUMMY_PROPOSAL_ID: u64 = u64::MAX;
+
+    for (scirpt_idx, (script_path, script_blob)) in scripts.iter().zip(compiled_scripts).enumerate()
+    {
+        let script_name = script_path.file_name().unwrap().to_string_lossy();
+        println!("    {}", script_name);
+
+        let vm = AptosVM::new(&state_view);
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        let (_vm_status, vm_output) = vm.execute_user_transaction(
+            &resolver,
+            &account
+                .account()
+                .transaction()
+                .script(Script::new(script_blob, vec![], vec![
+                    TransactionArgument::U64(DUMMY_PROPOSAL_ID), // dummy proposal id, ignored by the patched function
+                ]))
+                .chain_id(chain_id.chain_id())
+                .sequence_number(scirpt_idx as u64)
+                .gas_unit_price(gas_params.vm.txn.min_price_per_gas_unit.into())
+                .max_gas_amount(100000)
+                .ttl(u64::MAX)
+                .sign(),
+            &log_context,
+        );
+
+        let txn_output = vm_output.try_materialize_into_transaction_output(&resolver)?;
+
+        let txn_status = txn_output.status();
+        println!(
+            "{}",
+            format!("{:#?}", txn_status)
+                .lines()
+                .map(|line| format!("        {}", line))
+                .collect::<Vec<_>>()
+                .join("\n")
+        );
+        if !matches!(
+            txn_status,
+            TransactionStatus::Keep(ExecutionStatus::Success)
+        ) {
+            bail!("failed to execute governance script: {}", script_name)
+        }
+
+        let (write_set, _events) = txn_output.into();
```

Comment:
Recently we have been discussing with @runtian-zhou that we should always reconfig on governance proposal, especially if it is a framework upgrade (it is not the case now). I expect this simulation tool will be used a lot, should we check that events issue a new epoch event? Asserting `AptosVM::should_restart_execution` sufficies and will be a good place to catch problems.